### PR TITLE
fix(hierarchical grid): add richer data and a few new columns

### LINKED ISSUE #3250: Set all Grid samples width to be 100%
### Description  
Previously we had to set a fixed width in order for the Grid to fit the iframe, but now the iframe is sized based on the codeview container and its okay to set grid's height and its parent element to be 100% in order to take the full size within the code view and at the same time be with full screen size when opened alone. 

Example with CRM Grid:
image

Full screen:
image

### PR DESCRIPTION
**Pull request by monacodelisa**
Closes #3250

### CODE REVIEW

File: src/app/hierarchical-grid/hierarchical-grid-column-pinning/hierarchical-grid-toolbar-pinning.component.ts [hunk 1]
```diff
@@ -9,15 +9,16 @@ import { ActivatedRoute } from '@angular/router';
 })
 export class HGridToolbarPinningComponent implements OnInit {
     public localdata;
-    public dark = false;
+    public dark = "";
```
Comment (hanastasov) [inline src/app/hierarchical-grid/hierarchical-grid-column-pinning/hierarchical-grid-toolbar-pinning.component.ts:12]: I see why it is done this way (setting it as string) and it works out nicely when you receive a string value from the query params.

In general, IMO, it is good to keep the value boolean, and the check whether it is a "true" string being moved from the template to the place where you set the value on line 21:

```
this.dark = params.dark === 'true' ? true : false;
```

File: src/app/data/hierarchical-data.ts [hunk 2]
```diff
@@ -519,17 +543,26 @@ export const CUSTOMERS = [{
...
+            Discount: 5.0000001e-002,
+            Weight: 3.1,
+            Length: 24.3,
+            TotalPrice: 219.25
         }, {
             ProductID: 65,
             UnitPrice: 16.8000,
             Quantity: 20,
-            Discount: 0.0000000e+000
+            Discount: 0.0000000e+000,
```
Comment (hanastasov) [inline src/app/data/hierarchical-data.ts:562]: Why setting the discount in this format, seems curious

File: src/app/grid/grid-sample-pinning/grid-toolbar-pinning.component.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div [ngClass]="{'grid__wrapper': true, 'dark-theme': dark === true }" style='width: 100%;'>
+<div [ngClass]="{'grid__wrapper': true, 'dark-theme': dark === 'true' }" style='width: 100%;'>
```
Comment (hanastasov) [inline src/app/grid/grid-sample-pinning/grid-toolbar-pinning.component.html:1]: After keeping the `dark` value boolean, you will be able to use a simpler expression here:

```
<div [ngClass]="{'grid__wrapper': true, 'dark-theme': dark }" style='width: 100%;'>
```
Then it seems even nicer to rename the constant to smth more descrptive, for example `useDarkTheme`, and then the template seems even more readable:

```
<div [ngClass]="{'grid__wrapper': true, 'dark-theme': useDarkTheme }" style='width: 100%;'>
```

File: src/app/hierarchical-grid/hierarchical-grid-column-pinning/hierarchical-grid-toolbar-pinning.component.ts (hunk 1 above)
Comment (hanastasov) [reply src/app/hierarchical-grid/hierarchical-grid-column-pinning/hierarchical-grid-toolbar-pinning.component.ts:12]: > 

Null coalescing operator serves a more specific purpose, so when I think about it the Logical OR operator would serve us best here.

File: src/app/data/hierarchical-data.ts (hunk 2 above)
Comment (monacodelisa) [reply src/app/data/hierarchical-data.ts:562]: This is not a change applied by me I only added the properties below it

File: src/app/grid/grid-sample-pinning/grid-toolbar-pinning.component.html (hunk 3 above)
Comment (monacodelisa) [reply src/app/grid/grid-sample-pinning/grid-toolbar-pinning.component.html]: Ok I'm on it

**Review by hanastasov — APPROVED** [review]

**Pull request merged**